packages/client/src/generation/TSClient/PrismaClient.ts

```diff
@@ -33,10 +33,16 @@ function interactiveTransactionDefinition(this: PrismaClientClass) {
     return ''
   }
 
-  const txOptions = `{ maxWait?: number, timeout?: number }`
+  const txOptions = ['maxWait?: number', 'timeout?: number']
+
+  if (this.dmmf.hasEnumInNamespace('TransactionIsolationLevel', 'prisma')) {
+    txOptions.push('isolationLevel?: Prisma.TransactionIsolationLevel')
+  }
 
   return `
-  $transaction<R>(fn: (prisma: Prisma.TransactionClient) => Promise<R>, options?: ${txOptions}): Promise<R>;`
+  $transaction<R>(fn: (prisma: Prisma.TransactionClient) => Promise<R>, options?: {${txOptions.join(
+    ', ',
+  )}}): Promise<R>;`
```

Comment:
```suggestion
const options = `{${txOptions.join(", ")}}`;
  return `
    $transaction<R>(fn: (prisma: Prisma.TransactionClient) => Promise<R>, options?: ${options}): Promise<R>;
    `;
```

What do you think about defining the options above so its easier to read?